lib/batch_loader.rb

```diff
@@ -75,12 +77,20 @@ def __ensure_batched
     return if __executor_proxy.value_loaded?(item: @item)
 
     items = __executor_proxy.list_items
-    loader = ->(item, value) { __executor_proxy.load(item: item, value: value) }
+    loader =  -> (item, value = NULL_VALUE, &block) {
+      if block
```

Comment:
I had no idea you could pass a block to a lambda, that was the only reason I extracted this, so I have re-inlined it.